python/ray/ml/predictors/integrations/tensorflow/tensorflow_predictor.py

```diff
@@ -0,0 +1,123 @@
+from typing import Optional, Union, List, Type
+
+import tensorflow as tf
+
+from ray.ml.predictor import Predictor, DataBatchType
+from ray.ml.preprocessor import Preprocessor
+from ray.ml.checkpoint import Checkpoint
+
+# TensorFlow model objects cannot be pickled, therefore we use
+# a callable that returns the model, instead of a model object
+# itself.
+
+
+class TensorflowPredictor(Predictor):
+    """A predictor for TensorFlow models.
+
+    Args:
+        model_definition: A callable that returns a TensorFlow Keras model
+            to use for predictions.
+        preprocessor: A preprocessor used to transform data batches prior
+            to prediction.
+        model_weights: List of weights to use for the model.
+    """
+
+    def __init__(
+        self,
+        model_definition: Type[tf.keras.Model],
```

Comment:
Should this be Callable[[], tf.keras.Model] to match what’s in the doc string?

Reply:
I think this is equivalent. Type signifies an uninitialized class. I may clarify that further in the docstring

Reply:
I think we should err on the side of clarity...when I see this I think I have to pass in a class. But from the docstring, it seems like I have to pass in a function. Which one should I be doing?

Reply:
Anything that will return an initialized model after being called with no arguments. I can make it an union of the two?

Reply:
yes, I think that would clarify it!